### DISCUSSION TITLE
Is there a way to dispatch superclass methods?

### DISCUSSION THREAD

**Discussion by MordorianGuy** [coady/multimethod#75] [General]
I would like to dispatch the `__init__` method of a superclass. But it does not work with `multimethod` & `multimeta` in the way I expected. Moreover, manual registration is impossible as I cannot access the parent `__init__` method inside the class body. I have described more [here](https://stackoverflow.com/questions/74544561/how-to-dispatch-methods-of-a-superclass).

**Comment by coady** [comment]
To share a method across classes, the `register` decorator can be used. There's no direct equivalent of `super`, though it can be simulated by modifying or coercing the arguments.
```python
class Base:
    @multimethod
    def __init__(self, arg: int):
        self.arg = arg

class Subclass(Base):
    @Base.__init__.register
    def __init__(self, arg: bool):
        super().__init__(int(arg))

assert Subclass(True).arg == 1
```

**Comment by MordorianGuy** [comment]
Thank you very much for your answer! It has helped.

Although taking into account that a superclass is likely third-party and its methods are not multimethod objects, the resulting construction looks like this in my case.

```python
class Base:
    def __init__(self, arg: int):
        self.arg = arg

class Subclass(Base):
    @multimethod(Base.__init__).register
    def __init__(self, arg: bool):
        super().__init__(int(arg))

assert Subclass(True).arg == 1
```

Do you mind if I retell this in the initial **StackOverflow** question with the citation?

**Reply by coady** [reply]
> Do you mind if I retell this in the initial **StackOverflow** question with the citation?

Not at all.
